### PR TITLE
Support jump boxes for development environment.

### PR DESCRIPTION
**Pull request by squirrelsc**
Runbook example,

```yaml
dev:
  mock_tcp_ping: True
  jump_boxes:
    - private_key_file: $(admin_private_key_file)
      address: 20.236.23.33
    - private_key_file: $(admin_private_key_file)
      address: 20.236.23.40
```

### CODE REVIEW

File: lisa/schema.py [hunk 1]
```diff
@@ -1309,6 +1310,67 @@ def type_name(cls) -> str:
...
+        add_secret(self.username, PATTERN_HEADTAIL)
+        add_secret(self.password)
+        add_secret(self.private_key_file)
+
+        if not self.password and not self.private_key_file:
+            raise LisaException(
+                "at least one of password or private_key_file need to be set when "
+                "connecting"
+            )
+        elif not self.private_key_file:
```
Comment (mcgov) [inline lisa/schema.py:1337]: Would this disallow use of an empty password? It's terrible practice to use empty passwords and I am onboard with just not supporting that. If it's an undocumented requirement that's a side effect of an internal library we should make sure to warn at runtime and also document it.

**Review by mcgov — COMMENTED** [review]
Super excited to have jumpbox support! Only note is a question of updating documentation for password requirements

File: lisa/schema.py (hunk 1 above)
Comment (squirrelsc) [reply lisa/schema.py:1337]: If there is no password and no private key, the exception raises on Ln1333. This `elif` is to handle non-empty password. The next `else` of Ln 1341 to handle non-empty private key. So with them all, the empty password is not allowed.

File: lisa/util/shell.py [hunk 2]
```diff
@@ -499,6 +527,83 @@ def _purepath_to_str(
...
+            if is_trace_enabled:
+                jb_logger.debug(f"creating connection from source: {runbook} ")
+            client = paramiko.SSHClient()
+            client.set_missing_host_key_policy(paramiko.MissingHostKeyPolicy())
+            client.connect(
+                hostname=runbook.address,
+                port=runbook.port,
+                username=runbook.username,
+                password=runbook.password,
+                key_filename=runbook.private_key_file,
```
Comment (dsrivastavv) [inline lisa/util/shell.py:548]: Does all jumpbox need to have same key?

**Comment by dsrivastavv** [comment]
What was the need for adding this feature? Is this required for some testcase?

File: lisa/util/shell.py (hunk 2 above)
Comment (squirrelsc) [reply lisa/util/shell.py:548]: It doesn't need. The key can be configured for each jumpbox like address or port.

**Comment by squirrelsc** [comment]
> What was the need for adding this feature? Is this required for some testcase?

It's not required by test cases. I got feedback that some developers work from home, and need to bypass corp VPN. The jumpboxes are supported by ssh_config, but LISA doesn't use the ssh_config. So we need to implement it in LISA to support jumpbox scenarios.

**Pull request merged**